src/Altinn.AccessManagement.Core/Helpers/RightsHelper.cs

```diff
@@ -90,12 +101,23 @@ public static List<Detail> AnalyzeDelegationAccessReason(Right right)
                 if (roleAccessSources.Any())
                 {
                     string requiredRoles = string.Join(", ", roleAccessSources.SelectMany(roleAccessSource => roleAccessSource.PolicySubjects.SelectMany(policySubjects => policySubjects)));
+                    List<AttributeMatch> attributeMatches = new List<AttributeMatch>();
+                    var thing = roleAccessSources.SelectMany(roleAccessSource => roleAccessSource.PolicySubjects);
+                    foreach (List<PolicyAttributeMatch> attributeMatch in thing)
+                    {
+                        attributeMatches.Add(attributeMatch.First());
```

Comment:
Should AddRange in stead. Every relevant role attribute should be added not just the First

Reply:
Since we are doing this several times, you could make a helper for this mapping from PolicyAttribute lists to a single Attribute list

Reply:
Changed to "AddRange" and turned it into a helper function. Should I do this using the automapper instead?